### PR TITLE
feat(replacement): add messageformat monorepo replacement

### LINKED ISSUE #17830: Add replacement for `messageformt` npm package
### What would you like Renovate to be able to do?

Add a replacement for `messageformt` npm package, was renamed to `@messageformat/core`

https://github.com/messageformat/messageformat

### If you have any ideas on how this should be implemented, please tell us here.

😉 

### Is this a feature you are interested in implementing yourself?

Maybe

### PR DESCRIPTION
**Pull request by Gabriel-Ladzaretti**
## Context
Closes #17830

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] ran on a real [repository](https://github.com/ladzaretti/message-format-replacement/pulls)

### CODE REVIEW

**Review by viceice — COMMENTED** [review]
i think all message format rules should be in a single preset

File: lib/config/presets/internal/replacements.ts [hunk 1]
```diff
@@ -587,4 +587,53 @@ const mui: PresetTemplate = {
   title: 'material-ui-to-mui',
 };
 
-addPresets(presets, mui);
+const messageFormatCoreCli: PresetTemplate = {
+  description:
+    'The `messageformat` monorepo package naming scheme changed from `MessageFormat-{{package}}`-to-`@messageformat/{{package}}`.',
+  packageRules: {
```
Comment (viceice) [inline lib/config/presets/internal/replacements.ts:593]: this should be an array

File: lib/config/presets/internal/replacements.ts [hunk 2]
```diff
@@ -587,4 +587,48 @@ const mui: PresetTemplate = {
   title: 'material-ui-to-mui',
 };
 
-addPresets(presets, mui);
+const messageFormat: PresetTemplate[] = [
+  {
+    description:
```
Comment (viceice) [inline lib/config/presets/internal/replacements.ts:592]: please use one preset with multiple package rules

Comment (Gabriel-Ladzaretti) [reply lib/config/presets/internal/replacements.ts:592]: this is currently isnt supported by the `addPresets` function, i think ill open a new refactor PR for it, and add the replacements afterwards.
wdyt?

Comment (viceice) [reply lib/config/presets/internal/replacements.ts:592]: why not simply add to static list above?
i don't see a big benefit of abstracting the present generation 🤔

Comment (Gabriel-Ladzaretti) [reply lib/config/presets/internal/replacements.ts:592]: it saves a bit of code, in this case it isnt much i guess
e.g. on the left it is the abstracted preset while on the right is the auto generated replacements.

in this case it is 3(auto) vs 5(gen), but in `mui`'s case it was 1vs12 if i recall correctly.

image

**Comment by Gabriel-Ladzaretti** [comment]
Blocked by 
- #18124

**Review by viceice — APPROVED** [review]

**Pull request merged**